### ISSUE TITLE
Change container type in Model

### ISSUE THREAD

**Issue by davidanthoff** [julia-mpsge/MPSGE.jl#129]
Change the `_sectors` variable to be a dictionary rather than a list. The type signature would be `Symbol => Union{ScalarSector,IndexSector}` where the key is the name of the Sector. This is where I think you can eliminate the name from the structs. 

This would also allow accessing elements of the MPSGE model easier. As an unrefined example:

```
function Base.getindex(M::MPSGE.Model,key::Symbol)
    if key in keys(M._sectors)
        return M._sectors[key]
    elseif ...

    end
    error("Key $key not in MPSGE Model")
end
```

_Originally posted by @mitchphillipson in https://github.com/anthofflab/MPSGE.jl/issues/126#issuecomment-1708900880_

**Comment by davidanthoff** [comment]
I think we essentially have a global namespace within a model, so it might actually make more sense to just add one additional `Dict{Symbol,Any}` that has all the names used and then holds all the sectors, commodities etc in it. That dict would essentially play two roles: 1) prevent adding two things with duplicate names, and 2) enable the kind of `getindex` above.

We could then also just keep the current arrays per type, i.e. that would require very little change.

**Comment by mitchphillipson** [comment]
I think this is a good solution that can point towards bigger future changes.

**Comment by mitchphillipson** [comment]
I implemented this in #141. But I took it a step farther. I eliminated the individual variable arrays and added a function to extract specific variable types.

**Comment by mitchphillipson** [comment]
This is done
